src/battle_transition.c

```diff
@@ -0,0 +1,4346 @@
+
+    sTransitionStructPtr->VBlank_DMA++;
+    return FALSE;
+}
+
+static bool8 Phase2_Wave_Func3(struct Task *task)
+{
+    DmaStop(0);
+    sub_8149F84();
+    DestroyTask(FindTaskIdByFunc(Phase2Task_Wave));
+    return FALSE;
+}
+
+static void VBlankCB_Phase2_Wave(void)
+{
+    DmaStop(0);
+    VBlankCB_BattleTransition();
+    if (sTransitionStructPtr->VBlank_DMA != 0)
+        DmaCopy16(3, gUnknown_02038C28[0], gUnknown_02038C28[1], 320);
+    REG_WININ = sTransitionStructPtr->WININ;
+    REG_WINOUT = sTransitionStructPtr->WINOUT;
+    REG_WIN0V = sTransitionStructPtr->WIN0V;
+    DmaSet(0, gUnknown_02038C28[1], &REG_WIN0H, 0xA2400001);
+}
+
+static void Phase2Task_Sydney(u8 taskId)
+{
+    gTasks[taskId].tMugshotId = MUGSHOT_SYDNEY;
+    Phase2Task_MugShotTransition(taskId);
+}
+
+static void Phase2Task_Phoebe(u8 taskId)
+{
+    gTasks[taskId].tMugshotId = MUGSHOT_PHOEBE;
+    Phase2Task_MugShotTransition(taskId);
+}
+
+static void Phase2Task_Glacia(u8 taskId)
+{
+    gTasks[taskId].tMugshotId = MUGSHOT_GLACIA;
+    Phase2Task_MugShotTransition(taskId);
+}
+
+static void Phase2Task_Drake(u8 taskId)
+{
+    gTasks[taskId].tMugshotId = MUGSHOT_DRAKE;
+    Phase2Task_MugShotTransition(taskId);
+}
+
+static void Phase2Task_Champion(u8 taskId)
+{
+    gTasks[taskId].tMugshotId = MUGSHOT_CHAMPION;
+    Phase2Task_MugShotTransition(taskId);
+}
+
+static void Phase2Task_MugShotTransition(u8 taskId)
+{
+    while (sPhase2_Mugshot_Funcs[gTasks[taskId].tState](&gTasks[taskId]));
+}
+
+static bool8 Phase2_Mugshot_Func1(struct Task *task)
+{
+    u8 i;
+
+    sub_8149F08();
+    dp12_8087EA4();
+    Mugshots_CreateOpponentPlayerSprites(task);
+
+    task->tData1 = 0;
+    task->tData2 = 1;
+    task->tData3 = 239;
+    sTransitionStructPtr->WININ = 63;
+    sTransitionStructPtr->WINOUT = 62;
+    sTransitionStructPtr->WIN0V = 160;
+
+    for (i = 0; i < 160; i++)
+    {
+        gUnknown_02038C28[1][i] = 0xF0F1;
+    }
+
+    SetVBlankCallback(VBlankCB0_Phase2_Mugshots);
+
+    task->tState++;
+    return FALSE;
+}
+
+static bool8 Phase2_Mugshot_Func2(struct Task *task)
+{
+    s16 i, j;
+    u16 *dst1, *dst2;
+    const u16 *MugshotsMap;
+
+    MugshotsMap = sMugshotsTilemap;
+    sub_8149F58(&dst1, &dst2);
+    CpuSet(sUnknown_085B9AF0, dst2, 0xF0);
+    LoadPalette(sOpponentMugshotsPals[task->tMugshotId], 0xF0, 0x20);
+    LoadPalette(sPlayerMugshotsPals[gSaveBlock2Ptr->playerGender], 0xFA, 0xC);
+
+    for (i = 0; i < 20; i++)
+    {
+        for (j = 0; j < 32; j++, MugshotsMap++)
+        {
+            dst1[i * 32 + j] = *MugshotsMap | 0xF000;
+        }
+    }
+
+    EnableInterrupts(INTR_FLAG_HBLANK);
+
+    SetHBlankCallback(HBlankCB_Phase2_Mugshots);
+    task->tState++;
+    return FALSE;
+}
+
+static bool8 Phase2_Mugshot_Func3(struct Task *task)
+{
+    u8 i, r5;
+    u16* toStore;
+    s16 value;
+    s32 mergedValue;
+
+    sTransitionStructPtr->VBlank_DMA = FALSE;
+
+    toStore = gUnknown_02038C28[0];
+    r5 = task->tData1;
```

Comment:
Does *= 2 work? If so, use it as it's more legible

Reply:
nope, it must be lsl

Reply:
The compiler seriously turns *= 2 into a mul command? Usually it's smart enough to take multiplication by powers of two and make them lsl

Reply:
it messes up registers